### PR TITLE
Fix misaligned mock eth query types in controller tests

### PR DESCRIPTION
**Pull request by MajorLift**
## Explanation

Fixes `// @ts-expect-error Mock eth query does not fulfill type requirements` annotations throughout core repo.

## References

- Builds upon
  - https://github.com/MetaMask/eth-query/pull/21
  - https://github.com/MetaMask/core/pull/2028
- Related https://github.com/MetaMask/core/issues/1823
- Closes https://github.com/MetaMask/core/issues/2036

## Changelog

N/A

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/controller-utils/src/util.test.ts [hunk 1]
```diff
@@ -436,57 +437,61 @@ describe('util', () => {
...
           getBlockByHash(blockId: any, cb: any) {
             cb(null, { id: blockId });
           }
         }
-        // @ts-expect-error Mock eth query does not fulfill type requirements
-        const result = await util.query(new EthQuery(), 'getBlockByHash', [
-          '0x1234',
-        ]);
+        const result = await util.query(
+          new MockEthQuery({} as Provider),
```
Comment (mcmire) [inline packages/controller-utils/src/util.test.ts:446]: Hmm. Now that we have a fake provider, I'm wondering if a better way to achieve this without the type assertion would be something like this:

``` suggestion
        const ethQuery = new EthQuery(new FakeProvider());
        jest.spyOn(ethQuery, 'getBlockByHash').mockResolvedValue({ id: '0x1234' });
        const result = await util.query(
          ethQuery,
```

(We'll have to import FakeProvider at the top of course)

File: packages/gas-fee-controller/src/fetchBlockFeeHistory.test.ts [hunk 2]
```diff
@@ -33,23 +34,20 @@ function times<T>(n: number, fn: (n: number) => T): T[] {
 }
 
 describe('fetchBlockFeeHistory', () => {
-  const ethQuery = { eth: 'query' };
-
+  const mockEthQuery = new EthQuery({} as Provider);
```
Comment (mcmire) [inline packages/gas-fee-controller/src/fetchBlockFeeHistory.test.ts:37]: Similar to `util.test.ts`:

``` suggestion
  const ethQuery = new EthQuery(new FakeProvider());

File: packages/gas-fee-controller/src/fetchGasEstimatesViaEthFeeHistory.test.ts [hunk 3]
```diff
@@ -31,7 +32,8 @@ describe('fetchGasEstimatesViaEthFeeHistory', () => {
     number: new BN(1),
     baseFeePerGas: new BN(100_000_000_000),
   };
-  const ethQuery = {
+  const mockEthQuery = {
+    sendAsync: EthQuery.prototype.sendAsync,
     blockNumber: async () => latestBlock.number,
     getBlockByNumber: async () => latestBlock,
   };
```
Comment (mcmire) [inline packages/gas-fee-controller/src/fetchGasEstimatesViaEthFeeHistory.test.ts:39]: Similar as above — we can make a real EthQuery here using a fake provider:

``` suggestion
  let ethQuery: EthQuery;

  beforeEach(() => {
    ethQuery = new EthQuery(new FakeProvider());
    jest.spyOn(ethQuery, 'blockNumber').mockResolvedValue(latestBlock.number);
    jest.spyOn(ethQuery, 'getBlockByNumber').mockResolvedValue(latestBlock);
  });

File: packages/transaction-controller/src/utils/transaction-type.test.ts [hunk 4]
```diff
@@ -10,7 +12,7 @@ describe('determineTransactionType', () => {
   };
 
   it('returns a token transfer type when the recipient is a contract, there is no value passed, and data is for the respective method call', async () => {
-    class EthQuery {
+    class MockEthQuery extends EthQuery {
```
Comment (mcmire) [inline packages/transaction-controller/src/utils/transaction-type.test.ts:15]: Similar (you get the idea)

**Review by mcmire — COMMENTED** [review]
These tests aren't set up very well :( I'm curious whether `jest.mock` would be a better approach to mock `eth-query` rather than making fake versions. That would be a bigger refactor though, so for now I've tried to suggest changes that I hope wouldn't take too long to resolve.

File: packages/controller-utils/src/util.test.ts (hunk 1 above)
Comment (MajorLift) [reply packages/controller-utils/src/util.test.ts:446]: Is this the `FakeProvider` in `network-controller`? I'm getting this error:
```ts
error TS2345: Argument of type 'FakeProvider' is not assignable to parameter of type 'Provider'.
```
It's also currently not exported from `network-controller` so ideally we'd need to create a new release to add the export.

Comment (MajorLift) [reply packages/controller-utils/src/util.test.ts]: Here's my attempt at applying `FakeProvider`: https://github.com/MetaMask/core/pull/2037/commits/0a036847c84399602c360943a2319d8f54ac5932

Comment (mcmire) [reply packages/controller-utils/src/util.test.ts]: I had thought that FakeProvider was defined at the top level like FakeBlockTracker. We'd have to pull it out to use like this, so never mind :(

Comment (MajorLift) [reply packages/controller-utils/src/util.test.ts]: We could keep this PR open until that's done. `FakeProvider` seems like a definite improvement over `{} as Provider`.

Comment (mcmire) [reply packages/controller-utils/src/util.test.ts]: Okay, new PR created here: https://github.com/MetaMask/core/pull/2039

Comment (MajorLift) [reply packages/controller-utils/src/util.test.ts]: Didn't realize it would be that simple.😅 Just added a review.

Comment (mcmire) [reply packages/controller-utils/src/util.test.ts]: I've merged #2039 so these changes should now work.

Comment (MajorLift) [reply packages/controller-utils/src/util.test.ts]: Fixed all of the mock eth queries to use `FakeProvider`: https://github.com/MetaMask/core/pull/2037/commits/687d3eb0d633a825e9814ccf98d1b4aac1caf605

**Review by mcmire — APPROVED** [review]
Looks good!

**Pull request merged**